packages/wonder-stuff-sentry/src/kind-sentry-error.js

```diff
@@ -0,0 +1,170 @@
+                },
+                other: metadata,
+            },
+        });
+    }
+
+    get sentryData(): $ReadOnly<SentryData> {
+        // (1) We always ensure there is metadata
+        // (2) We know that the sentry data has the fields we need.
+        // (3) We know that the sentry data in the metadata only has the fields
+        //     we need.
+        // $FlowIgnore[incompatible-use] (1)
+        // $FlowIgnore[incompatible-return] (2)
+        // $FlowIgnore[incompatible-indexer] (3)
+        return this.metadata.sentry;
+    }
+
+    /**
+     * Get a stack of sentry data from this error and all causal errors.
+     *
+     * @returns {$ReadOnlyArray<$ReadOnly<SentryData>>} A returning a stack of any
+     * sentry data, with the most recent error's data at the top of the stack.
+     * An empty array if nothing is found.
+     */
+    getAllSentryData(): $ReadOnlyArray<$ReadOnly<SentryData>> {
+        // First, get the stack of data from all causal errors.
+        const getNearestKindSentryCause = (cause: ?Error): ?KindSentryError => {
+            // If the cause is a KindSentryError, then return it.
+            if (cause instanceof KindSentryError) {
+                return cause;
+            }
+            // If the cause is our base class, then keep looking.
+            if (cause instanceof KindError) {
+                return getNearestKindSentryCause(cause.cause);
+            }
+            // Otherwise, assume we don't have any more causal KindSentryErrors.
+            return null;
+        };
+
+        // Get the nearest KindSentryError cause.
+        const nearestCause = getNearestKindSentryCause(this.cause);
+
+        // Get the stack of data from that cause.
+        const causeDataStack = nearestCause?.getAllSentryData() ?? [];
+
+        // Now, include our own data.
+        return [this.sentryData, ...causeDataStack];
+    }
+
+    /**
+     * Get all the sentry data from this and causal errors into one object.
+     *
+     * Parses the error and all causal errors, merging the list of any
+     * SentryData found to produce a single SentryData object.
+     * For `tags` and `contexts`, existing values of inner (earlier) errors are
+     * overwritten by outer (later) errors. For `fingerprints`, the arrays are
+     * concatenated together.
+     * @returns {$ReadOnly<SentryData>} A single SentryData object combining
+     * all the SentryData from this and causal errors it contains.
+     */
+    getMergedSentryData(): $ReadOnly<SentryData> {
+        // First, get the stack of data from all causal errors and this one.
+        const sentryDataArray = this.getAllSentryData();
+
+        // Then, reduce the stack of data into a single object.
+        // Note that we use `reduceRight` so that the most recent data is at the
+        // top of the stack and takes precedence over the older data.
+        return sentryDataArray.reduceRight(sentryDataReducer, EmptySentryData);
```

Comment:
@lilli this plus the reducer does the same as the equivalent code and util methods in the mobile repo.

`reduceRight` reduces from the end of the array to the start, as we want.

Reply:
sweet 🍬